crowbar_framework/spec/controllers/api/upgrade_controller_spec.rb

```diff
@@ -58,17 +58,15 @@
     end
 
     it "initiates the upgrade of nodes" do
-      allow(NodeObject).to(
-        receive(:find).and_return([NodeObject.find_node_by_name("testing")])
-      )
+      allow(Api::Upgrade).to receive(:nodes).and_return(true)
```

Comment:
Sorry, but I'm struggling to understand this commit message.  Which test is already covered by which model?  And how can a model cover a test anyway?  Surely it's the other way around - a test covering a model?

And how does the second sentence about slow testing relate to this commit?

This kind of disconnect in understanding seems to be quite a common problem at the moment, and it's really slowing reviews down, for me anyway.  I've suggested more frequent pair programming as a way of addressing it, but I'm open to other ideas too.  What would you suggest?  BTW did you read [these articles on the topic](http://mailman.suse.de/mlarch/SuSE/cloud-devel/2016/cloud-devel.2016.09/msg00089.html)?  I think they make a lot of good points.